src/types/fetchers.ts

```diff
@@ -23,19 +21,9 @@ export type StringMultiProcessorResult = string | undefined;
 
 export type OnChainDataFetcherResult = string | number;
 
-// TODO: refactor this type
-export type FeedFetcherInterfaceResult =
-  | Promise<number | undefined>
-  | Promise<StringMultiProcessorResult[]>
-  | Promise<OnChainDataFetcherResult>
-  | Promise<CryptoCompareHistoFetcherResult[] | undefined>
-  | Promise<CryptoCompareMultiProcessorResult[]>
-  | Promise<SovrynPriceFetcherResult>
-  | Promise<OptionsEntries>;
-
 export interface FeedFetcherInterface {
   // eslint-disable-next-line @typescript-eslint/no-explicit-any
-  apply(params: any, timestamp?: number): FeedFetcherInterfaceResult;
+  apply(params: any, symbol: string, timestamp?: number): Promise<any>;
```

Comment:
@DZariusz this might be the most controversial change, let me know if you don't mind.

For this kind of interface there is not much a difference than using `any`.